### ISSUE TITLE
Direct Memory OOM issue during upgrade to 0.4.0

### ISSUE THREAD

**Issue by fx19880617** [apache/incubator-pinot#5610]
The table has a few no-dictionary columns.
After the upgrade, the servers are experiencing OOM issue even when querying `SELECT * FROM myTable`.

```
Caused by: java.lang.OutOfMemoryError: Direct buffer memory
        at java.nio.Bits.reserveMemory(Bits.java:694) ~[?:1.8.0_252]
        at java.nio.DirectByteBuffer.<init>(DirectByteBuffer.java:123) ~[?:1.8.0_252]
        at java.nio.ByteBuffer.allocateDirect(ByteBuffer.java:311) ~[?:1.8.0_252]
        at org.apache.pinot.core.io.reader.impl.ChunkReaderContext.<init>(ChunkReaderContext.java:38) ~[pinot-all-0.5.0-SNAPSHOT-jar-with-dependencies.jar:0.5.0-SNAPSHOT-01cdc55f514b10bee2d8108d9736d4b57c48b517]
        at org.apache.pinot.core.io.reader.impl.v1.VarByteChunkSingleValueReader.createContext(VarByteChunkSingleValueReader.java:93) ~[pinot-all-0.5.0-SNAPSHOT-jar-with-dependencies.jar:0.5.0-SNAPSHOT-01cdc55f514b10bee2d8108d9736d4b57c48b517]
        at org.apache.pinot.core.io.reader.impl.v1.VarByteChunkSingleValueReader.createContext(VarByteChunkSingleValueReader.java:32) ~[pinot-all-0.5.0-SNAPSHOT-jar-with-dependencies.jar:0.5.0-SNAPSHOT-01cdc55f514b10bee2d8108d9736d4b57c48b517]
        at org.apache.pinot.core.operator.docvalsets.SingleValueSet.<init>(SingleValueSet.java:35) ~[pinot-all-0.5.0-SNAPSHOT-jar-with-dependencies.jar:0.5.0-SNAPSHOT-01cdc55f514b10bee2d8108d9736d4b57c48b517]
        at org.apache.pinot.core.operator.blocks.SingleValueBlock.<init>(SingleValueBlock.java:41) ~[pinot-all-0.5.0-SNAPSHOT-jar-with-dependencies.jar:0.5.0-SNAPSHOT-01cdc55f514b10bee2d8108d9736d4b57c48b517]
        at org.apache.pinot.core.segment.index.datasource.BaseDataSource.getNextBlock(BaseDataSource.java:105) ~[pinot-all-0.5.0-SNAPSHOT-jar-with-dependencies.jar:0.5.0-SNAPSHOT-01cdc55f514b10bee2d8108d9736d4b57c48b517]
```
cc @siddharthteotia @Jackie-Jiang @kishoreg

**Comment by siddharthteotia** [comment]
@fx19880617 , do we have the segments handy? We can load the segment and look into the file header to see number of documents per chunk. I had made an improvement (as part of supporting text blobs for text index) by deriving the number of documents per chunk based on the length of data as opposed to hard coded 1000. So if we pack more than 1000 rows per chunk, the size of each chunk will increase. 

I don't think that would be the issue since I had this PR https://github.com/apache/incubator-pinot/pull/5470 which was included in the 0.4.0 release (a new RC was cut by Haibo to include this) and actually made this behavior configurable (disabled by default)

**Comment by fx19880617** [comment]
Segments are not handy and downloadable, I feel I need to write a program to run in prod then load segment and see the file headers. Those segments are created using 0.3.0 release.
